cmd/erasure-server-pool.go

```diff
@@ -1001,20 +1001,10 @@ func (z *erasureServerPools) PutObject(ctx context.Context, bucket string, objec
 	}
 
 	object = encodeDirObject(object)
-
 	if z.SinglePool() {
-		if !isMinioMetaBucketName(bucket) {
-			avail, err := hasSpaceFor(getDiskInfos(ctx, z.serverPools[0].getHashedSet(object).getDisks()...), data.Size())
-			if err != nil {
-				logger.LogOnceIf(ctx, err, "erasure-write-quorum")
-				return ObjectInfo{}, toObjectErr(errErasureWriteQuorum)
-			}
-			if !avail {
-				return ObjectInfo{}, toObjectErr(errDiskFull)
-			}
-		}
```

Comment:
Just wanted to check if this is intended. This does not seem in the scope of this PR.

Reply:
It is intended @vadmeste we do not have to do premature checks here.

Lower layers return appropriate errors properly when this happens.

We shouldn't really have to worry about drive space in this manner for every I/O it is quite expensive.

https://www.brendangregg.com/blog/2016-09-03/sudden-disk-busy.html

> All coming from stat(). The developers then realized they had enabled an app called "DiskUsageMonitor", which constantly stat()s the file system to track used space.

We shouldn't have to pay the price for something that will happen like 2yrs down the line.